src/evidently/v2/datasets.py

```diff
@@ -0,0 +1,129 @@
+import abc
+import dataclasses
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Union
+
+import pandas as pd
+
+from evidently import ColumnType
+from evidently.features.generated_features import GeneratedFeatures
+from evidently.options.base import Options
+
+
+@dataclasses.dataclass
+class ColumnInfo:
+    type: ColumnType
+
+
+class DataDefinition:
+    _columns: Dict[str, ColumnInfo]
+
+    def __init__(self, columns: Dict[str, ColumnInfo]):
+        self._columns = columns
+
+    def get_column_type(self, column_name: str) -> ColumnType:
+        # TODO: implement
+        return ColumnType.Unknown
+
+
+class DatasetColumn:
+    type: ColumnType
+    data: pd.Series
+
+    def __init__(self, type: ColumnType, data: pd.Series) -> None:
+        self.type = type
+        self.data = data
+
+
+class Scorer:
+    def __init__(self, alias: str):
+        self._alias = alias
+
+    @abc.abstractmethod
+    def generate_data(self, dataset: "Dataset") -> Union[DatasetColumn, Dict[str, DatasetColumn]]:
```

Comment:
I try to avoid unions, because every it means i will need to do instance check somewhere. How about we use general interface with a dict, but have something like `SingleColumnScorer` that will delegate this method to another abs method that returns a single  `DatasetColumn`?